org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/DateRangeEditor.java

```diff
@@ -0,0 +1,142 @@
+import org.adempiere.webui.component.Textbox;
+import org.adempiere.webui.component.ToolBarButton;
+import org.adempiere.webui.component.ZkCssHelper;
+import org.adempiere.webui.editor.WEditor;
+import org.adempiere.webui.event.ValueChangeEvent;
+import org.adempiere.webui.event.ValueChangeListener;
+import org.adempiere.webui.theme.ThemeManager;
+import org.zkoss.zk.ui.event.Events;
+import org.zkoss.zul.Div;
+
+/**
+*
+* @author Peter Takacs, Cloudempiere
+*
+*/
+public class DateRangeEditor extends Div implements ValueChangeListener {
+
+	/**
+	 *
+	 */
+	private static final long serialVersionUID = 3329360781729704243L;
+
+	private static final String IMAGES_CONTEXT_CALENDAR_PNG = "images/Calendar16.png";
+
+	private Textbox textbox;
+	ToolBarButton popupBtn;
+	private DateRangePicker popup;
+	private boolean isProcessParameter;
+	private WEditor editor;
+	private WEditor editor2;
+
+	/**
+	 * Constructor
+	 */
+	public DateRangeEditor(WEditor editor, WEditor editor2, boolean isProcessParameter) {
+		super();
+		this.isProcessParameter = isProcessParameter;
+		this.editor = editor;
+		this.editor2 = editor2;
+		init();
+	}
+
+	private void init() {
+
+		this.setWidth("100%");
+		ZkCssHelper.appendStyle(this, "display: inline-block; position: relative;");
+		textbox = new Textbox();
+		textbox.setStyle("width: 100%; background: white !important");
+		textbox.setReadonly(true);
+		this.appendChild(textbox);
+
+		popupBtn = new ToolBarButton();
+		popupBtn.addSclass("date-picker-calendar-button");
+		if (ThemeManager.isUseFontIconForImage())
+			popupBtn.setIconSclass("z-icon-Calendar");
+		else
+			popupBtn.setImage(ThemeManager.getThemeResource(IMAGES_CONTEXT_CALENDAR_PNG));
+		this.appendChild(popupBtn);
+
+		popup = new DateRangePicker(editor, editor2);
+		popupBtn.setTooltip(popup);
+		popupBtn.addEventListener(Events.ON_CLICK, event -> {
+			popup.setPage(popupBtn.getPage());
+			popup.open(popupBtn, "after_center");
+			LayoutUtils.autoDetachOnClose(popup);
+		});
+
+		Button btn = popup.getUpdateButton();
+		btn.addEventListener(Events.ON_CLICK, e -> {
+			textbox.setValue(popup.getDisplayValue());
+		});
+		editor.addValueChangeListener(this);
+		editor2.addValueChangeListener(this);
+		valueChange(null);
+	}
+
+	@Override
+	public void valueChange(ValueChangeEvent evt) {
+		textbox.setValue(popup.getDisplayValue());
+	}
+
+	/**
+	 * Set/Remove Mandatory Style
+	 * @param applyStyle
+	 */
+	public void setFieldMandatoryStyle() {
+		if (isMandatoryStyle())
+			LayoutUtils.addSclass("idempiere-mandatory", textbox);
+		else
+			LayoutUtils.removeSclass("idempiere-mandatory", textbox);
+	}
+
+	private boolean isMandatoryStyle() {
+		return editor.isMandatory() && editor.isReadWrite() && (isProcessParameter || editor.getGridField().isEditable(true)) && isNullOrEmpty(editor)
+				|| editor2.isMandatory() && editor2.isReadWrite() && (isProcessParameter || editor2.getGridField().isEditable(true)) && isNullOrEmpty(editor2);
```

Comment:
this is a strange one - why always mandatory if it is process parameter ?

Reply:
I think it is unnecessary, left there by mistake